src/collection-view.js

```diff
@@ -457,9 +460,13 @@ Marionette.CollectionView = Marionette.View.extend({
     return view;
   },
 
-  // check if the collection is empty
-  isEmpty: function() {
-    return !this.collection || this.collection.length === 0;
+  // check if the collection is empty, taking the filter into account unless 'skipFilter' is true
+  isEmpty: function(collection, options) {
```

Comment:
Alternatives to the given approach

1. Change this method to reference the collection that's passed in, and have  `_renderChildren` pass in a collection with the filtered models
2. Pass in an arbitrary array of models  as an option instead of `skipFilter`
3. Only consider the filter in `checkEmpty`

I'm thinking that option 1 might be the best, but I'm also concerned about how it might change the contract, if at all.